src/ArduinoGraphics.cpp

```diff
@@ -124,6 +124,51 @@ void ArduinoGraphics::noStroke()
   _stroke = false;
 }
 
+void ArduinoGraphics::ellipse(int x, int y, int width, int height)
+{
+  if (!_stroke && !_fill) {
+    return;
+  }
+
+  float theta;
+  int r1 = width / 2;
+  int r2 = height / 2;
+  for (int angle = 0; angle < 360; angle += 1) {
+    theta = angle * 3.14 / 180;
+    int xi = r1 * cos(theta);
+    int yi = r2 * sin(theta);
+    if (_stroke) {
+        // stroke
+        set(x + xi, y - yi, _strokeR, _strokeG, _strokeB);
+      } else if (_fill) {
```

Comment:
You are treating fill as an alternative stroke. The fill should fill the inside of the ellipse, so it needs to be done in addition to the stroke.

Reply:
fixed.

Reply:
Sorry still not fixed. I notice another bug with fill.
I am trying to solve it.